### PR TITLE
Tools 2790 gen conf

### PR DESCRIPTION
**Pull request by jdogmcsteezy**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,24 +3,30 @@ module github.com/aerospike/asconfig
 go 1.20
 
 require (
-	github.com/aerospike/aerospike-management-lib v0.0.0-20231106202816-b2438dbb7e03
-	github.com/bombsimon/logrusr/v4 v4.0.0
-	github.com/docker/docker v24.0.6+incompatible
-	github.com/go-logr/logr v1.2.4
+	github.com/aerospike/aerospike-client-go/v6 v6.14.1
+	github.com/aerospike/aerospike-management-lib v1.1.0
```
Comment (jdogmcsteezy) [inline go.mod:7]: Waiting on an update to the mgmt-lib

File: cmd/root.go [hunk 2]
```diff
@@ -47,22 +45,12 @@ func newRootCmd() *cobra.Command {
 
 			lvlCode, err := logrus.ParseLevel(lvl)
 			if err != nil {
-				logger.Errorf("Invalid log-level %s", lvl)
-				multiErr = fmt.Errorf("%w, %w %w", multiErr, errInvalidLogLevel, err)
+				// logger.Errorf("Invalid log-level %s", lvl)
```
Comment (dwelch-spike) [inline cmd/root.go:48]:
```suggestion
```
nit: I think this can be removed

File: conf/config_handler_mock.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	return o.confMap
+}
+
+func (o *mockCFG) ToConfFile() asconfig.DotConf {
+	return o.confText
+}
+
+func (o *mockCFG) GetFlatMap() *asconfig.Conf {
+	return o.flatConf
+}
```
Comment (dwelch-spike) [inline conf/config_handler_mock.go:35]: I don't think we should include mock files here. They could go in a mocks sub package like in https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/aerospikereceiver/cluster/mocks

File: conf/config_validator.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		mgmtLogger:  mgmtLogger,
+		version:     version,
+	}
+}
+
+// Validate validates the parsed configuration in ac against
+// the Aerospike schema matching ac.aerospikeVersion.
+// ValidationErrors is not nil if any errors during validation occur.
+// ValidationErrors Error() method outputs a human readable string of validation error details.
+// error is not nil if validation, or any other type of error occurs.
```
Comment (dwelch-spike) [inline conf/config_validator.go:35]: This comment needs to be updated to match the current state of the file.
For example, there is no more `ac.aerospikeVersion`.
The lines about `ValidationError` Error() method should be moved to that method.

**Review by dwelch-spike — APPROVED** [review]
Just a few nitpicks. Otherwise looks good!

File: conf/config_handler_mock.go (hunk 3 above)
Comment (dwelch-spike) [reply conf/config_handler_mock.go:35]: not needed

**Comment by dwelch-spike** [comment]
Note: The Snyk report points to a low vulnerability issue with no fix yet. I don't think we need to block on it.

**Comment by dwelch-spike** [comment]
Review changes LGTM

**Pull request merged**